proxy/src/transport/tls/config.rs

```diff
@@ -100,6 +100,11 @@ pub enum ReasonForNoTls {
     /// The connection isn't TLS or it is TLS but not intended to be handled
     /// by the proxy.
     NotProxyTls,
+
+    /// We fell back to plaintext because the TLS handshake failed.
+    // TODO: Perhaps this should store some more detailed information on
+    // why the handshake failed...
```

Comment:
What are the reasons that a handshake can fail? What do we need to do get access to this information?

Reply:
This comment is essentially intended as a placeholder to say "I'm working on this in another branch" (#1212 does this). I can remove it if you prefer.

Reply:
i'm fine with it if there's an immediate followup